### PR TITLE
Validate: add ability to expand `src validate` commands for future health check functionality

### PR DESCRIPTION
**Pull request by jdpleiness**
To support expansion of the `src validate` command with new health check features, the validate command needs to be broken down into subcommands.

This PR moves the current `src validate` command to `src validate install` as this command validates a Sourcegraph installation is setup and working correctly. Additional changes and features are listed below:

* Switch to using existing `internal/api` package for  GraphQL queries
* Remove _create first admin_ feature that allowed the validate command to create a Sourcegraph admin user
* Refactor configuration layout to allow easier support for multiple code hosts in the future. `src validate` currently only supports GitHub, but adding additional code hosts such as GitLab is now much easier.
* Add visual feedback to CLI for success or failure of steps
* Add cleanup of insight testing as mentioned [here](https://github.com/sourcegraph/src-cli/pull/912#issuecomment-1377084768)
* Move GitHub token from config file to an environmental variable in alignment with other account tokens

validate-install

validate-install-yaml

validate-install-json

TODO:
- [x] Update CHANGELOG.md

### Test plan

Locally tested changes using latest Sourcegraph Helm install. Tested default configuration if no YAML is provided, YAML configuration provided, and JSON configuration provided.

* ran `go test ./...` - all tests passing
* ran `staticcheck ./...` - no issues found in new code

<details><summary>Example YAML file used:</summary>

```yaml
externalService:
  kind: GITHUB
  displayName: srcgraph-test
  deleteWhenDone: true
  maxRetries: 5
  retryTimeoutSeconds: 5
  config:
    gitHub:
      url: https://github.com
      orgs: []
      repos:
        - sourcegraph-testing/zap

searchQuery:
  - repo:^github.com/sourcegraph-testing/zap$ test
  - repo:^github.com/sourcegraph-testing/zap$@v1.14.1 test

insight:
  title: "Javascript to Typescript migration"
  dataSeries:
    [ {
      "query": "lang:javascript",
      "label": "javascript",
      "repositoryScope": [
        "github.com/sourcegraph/sourcegraph"
      ],
      "lineColor": "#6495ED",
      "timeScopeUnit": "MONTH",
      "timeScopeValue": 1
    },
      {
        "query": "lang:typescript",
        "label": "typescript",
        "lineColor": "#DE3163",
        "repositoryScope": [
          "github.com/sourcegraph/sourcegraph"
        ],
        "timeScopeUnit": "MONTH",
        "timeScopeValue": 1
      }
    ]
```

</details>

<details><summary>Example JSON file used:</summary>

```json
{
  "externalService": {
    "kind": "GITHUB",
    "displayName": "srcgraph-test",
    "deleteWhenDone": true,
    "maxRetries": 5,
    "retryTimeoutSeconds": 5,
    "config": {
      "gitHub": {
        "url": "https://github.com",
        "orgs": [],
        "repos": [
          "sourcegraph-testing/zap"
        ]
      }
    }
  },

  "searchQuery": [
    "repo:^github.com/sourcegraph-testing/zap$ test",
    "repo:^github.com/sourcegraph-testing/zap$@v1.14.1 test"
  ],

  "insight": {
    "title": "Javascript to Typescript migration",
    "dataSeries": [
      {
        "query": "lang:javascript",
        "label": "javascript",
        "repositoryScope": [
          "github.com/sourcegraph/sourcegraph"
        ],
        "lineColor": "#6495ED",
        "timeScopeUnit": "MONTH",
        "timeScopeValue": 1
      },
      {
        "query": "lang:typescript",
        "label": "typescript",
        "lineColor": "#DE3163",
        "repositoryScope": [
          "github.com/sourcegraph/sourcegraph"
        ],
        "timeScopeUnit": "MONTH",
        "timeScopeValue": 1
      }
    ]
  }

}
```

</details>

### CODE REVIEW

File: cmd/src/validate_install.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+			if err != nil {
+				return err
+			}
+		}
+
+		if validationSpec == nil {
+			validationSpec = validate.DefaultConfig()
+		}
+
+		envGithubToken := os.Getenv("SRC_GITHUB_TOKEN")
```
Comment (eseliger) [inline cmd/src/validate_install.go:98]: should we warn if this is not provided?

**Review by eseliger — APPROVED** [review]

File: cmd/src/validate_install.go (hunk 1 above)
Comment (jdpleiness) [reply cmd/src/validate_install.go:98]: Good call. I'll add a warning and wrap it in some nice output for the user if the env var isn't there.

**Review by abeatrix — APPROVED** [review]
I like how nice the output looks!
Idea: In addition to the current output, I think it'd be nice to show the services each task are related to, so that if one of test fails, the user will know what to look into?

File: internal/validate/install.go [hunk 2]
```diff
@@ -0,0 +1,476 @@
...
+	ExternalService ExternalService `yaml:"externalService"`
+
+	// Insight used for validation testing.
+	Insight Insight `yaml:"insight"`
+}
+
+// DefaultConfig returns a default configuration to be used for testing.
+func DefaultConfig() *ValidationSpec {
+	return &ValidationSpec{
+		SearchQuery: []string{"repo:^github\\.com/gorilla/mux$ Router", "repo:^github\\.com/gorilla/mux$@v1.8.0 Router"},
```
[the remark this answers is not in the document]
Comment (jdpleiness) [reply internal/validate/install.go:82]: Yeah, thats a good idea!

**Review by abeatrix — CHANGES_REQUESTED** [review]
I just noticed https://github.com/gorilla/mux has been archive so the search queries will no longer work. We might want to use another repo for default

File: internal/validate/install.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jdpleiness) [reply internal/validate/install.go]: Just added a symbol search to the defaults, should be gtg.

**Review by abeatrix — APPROVED** [review]
yay thanks for the quick update! LGTM!

**Review by DaedalusG — COMMENTED** [review]
Late to the party but this looks great!

**Pull request merged**